unit-tests/src/test/scala/scala/scalanative/posix/TimeTest.scala

```diff
@@ -286,10 +286,11 @@ class TimeTest {
       // time zone, so the check would fail if we parse a date with a
       // different time zone.
 
-      val cp = if (Platform.isFreeBSD)
-        strptime(c"Fri Mar 31 14:47:44 2017", c"%a %b %d %T %Y", tmPtr)
-      else
-        strptime(c"Fri Mar 31 14:47:44 GMT 2017", c"%a %b %d %T %Z %Y", tmPtr)
+      val cp =
+        if (Platform.isFreeBSD)
+          strptime(c"Fri Mar 31 14:47:44 2017", c"%a %b %d %T %Y", tmPtr)
```

Comment:
In the last two weeks or so there has been an update to TimeTest.scala
to allow it to additionally run on macOS.  TimeTest, and this section in
particular, is a recurring pain.

Do you know if strptime() on FreeBSD  either:
1) rejects %Z (implied posix behavior).
2) parses %Z and does nothing (gnu linux behavior)
3) parses %Z and sets C fields (macOS behavior)

If #1, then my scars tell me putting the "if (Platform.isFreeBSD)" at
 the beginning of the SN mainline TimeTest.scala test and skipping
 the entire thing is the way to go.  

If either  #2 or #3, the current SN mainline test should work, as is,
and not require any special FreeBSD handling. (Unless I am missing
something, I have been known to do that.)

If there is some other behavior, please let me know (or, I suggest, just
 bypass as in case #1 above).

Does this help?

Reply:
As I added in the comment, FreeBSD parses %Z and use it to create the equivalent time in the local timezone. This means that if the string contains GMT and my local timezone is CET (or CEST, given that Mar 31 is in DST), the hour check fails because 16 != 14.

Reply:
@alexdupre 

1) Sorry about getting your name wrong, as the actor famously said
    to the reporter: Print anything you want about me but spell my name right!

2) Given your info about %Z parsing, I will go back to the comment you mention.
    I have spent __way__ too much of my life on this test over the past two years
    or more. 

    The problematic test is not really about strptime() but about showing that
    SN is using structure copy-out/copy-in with its "short" POSIX 'struct tm'
     declaration.   

     I think that test is passing on macOS CI because the local time on that
     system is UTC.  That, plus what you are seeing means that at least
     the hours & minutes part of the current test are __wrong__ in the wild.
     I will have to PR that. I will probably wait until after this PR is merged.
     Thank you for detecting that flaw.
     
Reviewers:  I believe fixing the pre-exiting TimeTest.scala flaw is on me and should
                   not hold up this PR.
                   this PR.

Reply:
Yes, surely if the system is in UTC the issue cannot be seen. I don't know what are the linux and macos behaviors, the linux man page is not clear enough to me, while on FreeBSD it clearly says that `The resulting values will be relative to local time`, and I can confirm it. But as you said, the test is not really about strptime.